### DISCUSSION TITLE
2d light system to make game have day and night effect

### DISCUSSION THREAD

**Discussion by jcy1001** [axmolengine/axmol#1297] [Q&A]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by rh101** [comment]
Isn't this a feature that is game/project specific? It really isn't something that belongs in the game engine.  If you need suggestions or help on how to implement such a thing, then there are more appropriate places to be asking such questions, such as the [Discussions ](https://github.com/axmolengine/axmol/discussions) section.

**Comment by halx99** [comment]
I guess you need: https://www.codeandweb.com/spriteilluminator/tutorials/how-to-use-light-effects-with-axmol-game-engine

**Comment by DelinWorks** [comment]
it's typically done with a fragment shader that is run on all of your sprites that are light affected, that shader can have an array of structs that explain where and how strong the lights are, but that's bad since you only have as many lights as the uniform size of your gpu can support, a better approach is to use render textures, your light objects can be sprites that have a shader, that shader doesn't need any texture but just a circular gradiant where in the middle it's white and the further it is from the center the more transparent pixels are, and then render your main sprites on another render texture, and overlay the two render textures by subtracting rt1 from rt2 on a third texture to get lighting, you can even exclude elements from lighting by not rendering them to the second render texture, what I just described is differed rendering for 2D, it's much more complex for 3D, the second technique of using render textures gives the benefits of batching since sprites are just quads with custom shaders that act like light. hope this helps.

**Reply by DelinWorks** [reply]
also, for a day night effect, you can darken the render texture's colors and tint them to blue (using a shader) and that should give you an idea of how nights look
